src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -291,10 +365,325 @@ def _amplitude_scalings_chunk(segment_index, start_frame, end_frame, worker_ctx)
-            templates.append(template)
+
             linregress_res = linregress(template.flatten(), local_waveform.flatten())
-            scalings.append(linregress_res[0])
-    scalings = np.array(scalings)
+            scalings[spike_index] = linregress_res[0]
+
+        # deal with collisions
+        if len(collisions) > 0:
+            num_spikes_in_previous_segments = int(
+                np.sum([len(spikes[segment_slices[s]]) for s in range(segment_index)])
+            )
+            for spike_index, collision in collisions.items():
+                scaled_amps = fit_collision(
+                    collision,
+                    traces_with_margin,
+                    start_frame,
+                    end_frame,
+                    left,
+                    right,
+                    nbefore,
+                    all_templates,
+                    unit_inds_to_channel_indices,
+                    cut_out_before,
+                    cut_out_after,
+                )
+                # the scaling for the current spike is at index 0
+                scalings[spike_index] = scaled_amps[0]
+
+                # make collision_dict indices "absolute" by adding i0 and the cumulative number of spikes in previous segments
+                collisions_dict.update({spike_index + i0 + num_spikes_in_previous_segments: collision})
+    else:
+        scalings = np.array([])
+        collisions_dict = {}
+
+    return (scalings, collisions_dict)
 
-    return (scalings,)
+
+### Collision handling ###
+def _are_unit_indices_overlapping(unit_inds_to_channel_indices, i, j):
+    """
+    Returns True if the unit indices i and j are overlapping, False otherwise
+
+    Parameters
+    ----------
+    unit_inds_to_channel_indices: dict
+        A dictionary mapping unit indices to channel indices
+    i: int
+        The first unit index
+    j: int
+        The second unit index
+
+    Returns
+    -------
+    bool
+        True if the unit indices i and j are overlapping, False otherwise
+    """
+    if len(np.intersect1d(unit_inds_to_channel_indices[i], unit_inds_to_channel_indices[j])) > 0:
+        return True
+    else:
+        return False
+
+
+def find_collisions(spikes, spikes_w_margin, delta_collision_samples, unit_inds_to_channel_indices):
+    """
+    Finds the collisions between spikes.
+
+    Parameters
+    ----------
+    spikes: np.array
+        An array of spikes
+    spikes_w_margin: np.array
+        An array of spikes within the added margin
+    delta_collision_samples: int
+        The maximum number of samples between two spikes to consider them as overlapping
+    unit_inds_to_channel_indices: dict
+        A dictionary mapping unit indices to channel indices
+
+    Returns
+    -------
+    collision_spikes_dict: np.array
+        A dictionary with collisions. The key is the index of the spike with collision, the value is an
+        array of overlapping spikes, including the spike itself at position 0.
+    """
```

Comment:
I am OK with this but the algortithm is a bit bru-force no ?
we are looping on all spikes and we are doing 2 searchsorted.

Algos like this https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/postprocessing/correlograms.py#L237
or this done @DradeAW 
https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/postprocessing/correlograms.py#L294
avoid the full double loop over spikes.

Algos to find collision is really close to computing correlogram

We can keep this for now but I would add a TODO for later: improve the loop.

Reply:
ok, later for improvement.